survey/default.nix

```diff
@@ -263,6 +265,17 @@ let
     # https://github.com/nh2/static-haskell-nix/issues/6#issuecomment-420494800
     "sparkle"
 
+    # PostgreSQL's test suite doesn't pass:
+    # https://github.com/NixOS/nixpkgs/issues/150930
+    #
+    # Even if the test suite is disabled, these Haskell binaries fail in
+    # linking with errors like:
+    #
+    # ld: libpq.a(fe-auth-scram.o): in function `pg_fe_scram_build_secret':
+    #   /build/postgresql-13.4/src/interfaces/libpq/fe-auth-scram.c:839:0: error:
+    #      undefined reference to `pg_saslprep'
+    "hasql-notifications" "hasql-queue" "postgresql-orm" "postgrest" "tmp-postgres"
```

Comment:
This is kind of unfortunate.

I created an issue in Nixpkgs, but it sounds like the alpine developers also had a bunch of trouble getting postgres-13 building: https://github.com/NixOS/nixpkgs/issues/150930

I guess we could do something like actively use `postgres_12` instead of 13 here in static-haskell-nix, but it would be great to just do that in a separate PR for someone that is interested in getting PostgreSQL working.

Reply:
There are still some stackage executables that depend on the test-failing postgres.

I think we should do what you say, and override postgres to be `postgres_12` until nixpkgs has a fix for that.

That should be OK especially because Haskell packages only really use the `libpq` client part of postgres, which should be fine on older versions.

Reply:
Ah, sorry for this, it looks like there was one haskell package (`persistent-postgresql`?) that did build with postgresql-13 as long as its tests were disabled.  Although since I didn't end up disabling the tests for postgres in this PR, I forgot to also add `persistent-postgresql` to the list of blacklisted packages!

---------------------------------------------------------

I took your advice and overrode `postgresql` to be `postgresql_11` in 3d0be9a.  Now all the stackage executables that use libpq are compiling correctly!

I first tried `postgresql_12`, but I was seeing strange linking errors with that as well.  It would be nice to use a more recent version of postgresql than 11, but hopefully we can just put that off for a future PR.